horizons/world/managers/productionfinishediconmanager.py

```diff
@@ -152,4 +154,4 @@ def get_resource_string(self, instance, res):
 		This key MUST be unique!
 		"""
 		return "produced_resource_" + str(res) + "_" + str(instance.position.origin)\
-		       + "_" + str(Scheduler().cur_tick)
+		       + "_" + str(uuid.uuid1())
```

Comment:
Has this any consequences for multiplayer? Shouldn't, right? Just asking, because this takes the current time. Don't know if the scheduling ticks are synced, though.

And reading the documentation, you might want to add parameters to `uuid1()`.

> Generate a UUID from a host ID, sequence number, and the current time. If node is not given, getnode() is used to obtain the hardware address. If clock_seq is given, it is used as the sequence number; otherwise a random 14-bit sequence number is chosen.

https://docs.python.org/2/library/uuid.html#uuid.getnode <- says, that it might be slow and I don't see, how the MAC address is relevant here. You could also use the `Scheduler().cur_tick` as sequence number, I guess.